### PR TITLE
JBPM-8577 Add more unit tests for All Validation Errors (BAPL-1217)

### PR DESCRIPTION
**Pull request by tiagodolphine**
Add more unit tests for All Validation Errors related to [BAPL-1217](https://issues.jboss.org/browse/BAPL-1217) 

@romartin 
@LuboTerifaj

### CODE REVIEW

**Comment by tiagodolphine** [comment]
Jenkins execute full downstream build

File: kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-backend-common/src/test/java/org/kie/workbench/common/stunner/core/backend/service/ValidationServiceImplTest.java [hunk 1]
```diff
@@ -103,20 +124,48 @@ public void validate(Diagram entity, Consumer<Collection<DomainViolation>> resul
...
+        });
+
+        DomainViolation first = violations.get(0);
+        when(first.getUUID()).thenReturn(UUID_0);
+
+        DomainViolation last1 = violations.get(violations.size() - 2);
+        when(last1.getUUID()).thenReturn("null");
+
+        DomainViolation last = violations.get(violations.size() - 1);
+        when(last.getUUID()).thenReturn(null);
```
[the remark this answers is not in the document]
Comment (tiagodolphine) [reply kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-backend-common/src/test/java/org/kie/workbench/common/stunner/core/backend/service/ValidationServiceImplTest.java]: there are 2 tests, one with `null` and other with a String with value `"null"`, there are some cases that this was happening during my manual tests, so I inserted a test for it.

File: kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-backend-common/src/test/java/org/kie/workbench/common/stunner/core/backend/service/ValidationServiceImplTest.java [hunk 2]
```diff
@@ -103,20 +124,48 @@ public void validate(Diagram entity, Consumer<Collection<DomainViolation>> resul
...
+        violations.stream().forEach(v -> {
+            when(v.getViolationType()).thenReturn(Violation.Type.ERROR);
+            when(v.getUUID()).thenReturn(UUID_1);
+        });
+
+        DomainViolation first = violations.get(0);
+        when(first.getUUID()).thenReturn(UUID_0);
+
+        DomainViolation last1 = violations.get(violations.size() - 2);
+        when(last1.getUUID()).thenReturn("null");
```
[the remark this answers is not in the document]
Comment (tiagodolphine) [reply kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-backend-common/src/test/java/org/kie/workbench/common/stunner/core/backend/service/ValidationServiceImplTest.java]: there are 2 tests, one with `null` and other with a String with value `"null"`, there are some cases that this was happening during my manual tests, so I inserted a test for it.

**Comment by tiagodolphine** [comment]
Hi @LuboTerifaj about your comments, the classes you mentioned were covered by tests for the modifications I made on the PR https://github.com/kiegroup/kie-wb-common/pull/2661, anyway can you be more specific on the test cases you think it is better to be covered?

- **BPMNValidatorImpl** is being tested, on the test class I added a new .bpmn file to test the modifications see

https://github.com/kiegroup/kie-wb-common/blob/fc9288877eb7d211bd3a0f67197d418ffa5e4402/kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/test/java/org/kie/workbench/common/stunner/bpmn/backend/validation/BPMNValidatorImplTest.java#L48
https://github.com/kiegroup/kie-wb-common/blob/fc9288877eb7d211bd3a0f67197d418ffa5e4402/kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/test/java/org/kie/workbench/common/stunner/bpmn/backend/validation/BPMNValidatorImplTest.java#L95

- **AbstractModelBeanValidator** there were added tests for the methods inserted on the PR, see

https://github.com/kiegroup/kie-wb-common/blob/fc9288877eb7d211bd3a0f67197d418ffa5e4402/kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-core-common/src/test/java/org/kie/workbench/common/stunner/core/validation/impl/ModelBeanValidatorTest.java#L107

-  **AbstractDiagramValidator** there were added tests for the methods inserted on the PR, see

https://github.com/kiegroup/kie-wb-common/blob/fc9288877eb7d211bd3a0f67197d418ffa5e4402/kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-core-common/src/test/java/org/kie/workbench/common/stunner/core/validation/impl/DiagramValidatorlTest.java#L121

https://github.com/kiegroup/kie-wb-common/blob/fc9288877eb7d211bd3a0f67197d418ffa5e4402/kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-core-common/src/test/java/org/kie/workbench/common/stunner/core/validation/impl/DiagramValidatorlTest.java#L177


- **ClientTranslationService** there were added tests for the methods inserted on the PR, see

https://github.com/kiegroup/kie-wb-common/blob/fc9288877eb7d211bd3a0f67197d418ffa5e4402/kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/test/java/org/kie/workbench/common/stunner/core/client/i18n/ClientTranslationServiceTest.java#L92

So, IMO these tests are covering the modifications, if you find more specific scenarios you can give more details and I can insert them to increase the coverage. On the other hand, I don't think it is the scope of this PR insert tests for code that was not modified by the PR https://github.com/kiegroup/kie-wb-common/pull/2661.

Thanks for the review again!

File: kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-backend-common/src/test/java/org/kie/workbench/common/stunner/core/backend/service/ValidationServiceImplTest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (LuboTerifaj) [reply kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-backend-common/src/test/java/org/kie/workbench/common/stunner/core/backend/service/ValidationServiceImplTest.java]: In `domainViolationList` at index `violations.size() - 2` there is `domainViolationNull` and not `domainViolationNullStr`

File: kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-backend-common/src/test/java/org/kie/workbench/common/stunner/core/backend/service/ValidationServiceImplTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (LuboTerifaj) [reply kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-backend-common/src/test/java/org/kie/workbench/common/stunner/core/backend/service/ValidationServiceImplTest.java]: In `domainViolationList` at `index violations.size() - 1` there is `domainViolationNullStr` and not `domainViolationNull`

**Comment by LuboTerifaj** [comment]
Hello @tiagodolphine ,

please, see my two comments in the last review.

Regarding the test coverage, see comments below:

- In **BPMNValidatorImpl** class I meant to cover https://github.com/kiegroup/kie-wb-common/blob/bb81e19585c4fc6f3a55a25b4bcc312ee15df375/kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/main/java/org/kie/workbench/common/stunner/bpmn/backend/validation/BPMNValidatorImpl.java#L116, but I am not sure whether we want to cover also exceptions, so I am okay with it.

- In **ClientTranslationService** class there are uncovered all methods except `getElementName()`, but yeah you are right, it is out of scope of this PR.

- Regarding **AbstractModelBeanValidator** there is uncovered https://github.com/kiegroup/kie-wb-common/blob/bb81e19585c4fc6f3a55a25b4bcc312ee15df375/kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-core-common/src/main/java/org/kie/workbench/common/stunner/core/validation/impl/AbstractModelBeanValidator.java#L50

- In **AbstractDiagramValidator** there is uncovered https://github.com/kiegroup/kie-wb-common/blob/bb81e19585c4fc6f3a55a25b4bcc312ee15df375/kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-core-common/src/main/java/org/kie/workbench/common/stunner/core/validation/impl/AbstractDiagramValidator.java#L89

Does it make sense to cover these lines or not?

Thank you!

File: kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-backend-common/src/test/java/org/kie/workbench/common/stunner/core/backend/service/ValidationServiceImplTest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (tiagodolphine) [reply kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-backend-common/src/test/java/org/kie/workbench/common/stunner/core/backend/service/ValidationServiceImplTest.java]: Ok, I'll change it.

File: kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-backend-common/src/test/java/org/kie/workbench/common/stunner/core/backend/service/ValidationServiceImplTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (tiagodolphine) [reply kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-backend-common/src/test/java/org/kie/workbench/common/stunner/core/backend/service/ValidationServiceImplTest.java]: Ok, I'll change it.

**Comment by tiagodolphine** [comment]
Jenkins execute full downstream build

**Comment by tiagodolphine** [comment]
Hi @LuboTerifaj I agreed with the comments I added more tests to increase the coverage, but in fact, most of them were not related to changes from the original PR #2661. Can you please take a look at the changes?

Thanks!

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/test/java/org/kie/workbench/common/stunner/bpmn/backend/validation/BPMNValidatorImplTest.java [hunk 3]
```diff
@@ -42,8 +44,10 @@
 @RunWith(MockitoJUnitRunner.class)
 public class BPMNValidatorImplTest {
 
-    private static final String BPMN_NO_END = "org/kie/workbench/common/stunner/bpmn/backend/service/diagram/validation/no-end.bpmn";
+    private static final String WRONG_PATH = "wrongPath";
```
Comment (LuboTerifaj) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/test/java/org/kie/workbench/common/stunner/bpmn/backend/validation/BPMNValidatorImplTest.java]: It is never used.

Comment (tiagodolphine) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/test/java/org/kie/workbench/common/stunner/bpmn/backend/validation/BPMNValidatorImplTest.java]: done

**Comment by LuboTerifaj** [comment]
Jenkins retest please.

**Comment by LuboTerifaj** [comment]
Jenkins execute full downstream build

**Comment by tiagodolphine** [comment]
Jenkins retest please.

**Comment by tiagodolphine** [comment]
Jenkins execute full downstream build

**Comment by tiagodolphine** [comment]
Jenkins retest please.

**Comment by tiagodolphine** [comment]
Jenkins execute full downstream build

**Comment by tiagodolphine** [comment]
Jenkins retest please.

**Comment by tiagodolphine** [comment]
Jenkins execute full downstream build

**Comment by tiagodolphine** [comment]
Jenkins retest please.

**Comment by tiagodolphine** [comment]
Jenkins execute full downstream build

**Comment by tiagodolphine** [comment]
Jenkins retest please.

**Comment by tiagodolphine** [comment]
Jenkins execute full downstream build

**Comment by tiagodolphine** [comment]
Jenkins retest please.

**Comment by tiagodolphine** [comment]
Jenkins execute full downstream build

**Comment by cristianonicolai** [comment]
jenkins please retest this

**Comment by tiagodolphine** [comment]
Jenkins execute full downstream build

**Comment by tiagodolphine** [comment]
@romartin @LuboTerifaj it is green again, so it is ok to be merged.

**Pull request merged**